src/main/kotlin/me/moeszyslak/logbot/listeners/VoiceListener.kt

```diff
@@ -0,0 +1,35 @@
+package me.moeszyslak.logbot.listeners
+
+import com.gitlab.kordlib.core.behavior.channel.createEmbed
+import com.gitlab.kordlib.core.entity.channel.TextChannel
+import com.gitlab.kordlib.core.event.VoiceStateUpdateEvent
+import me.jakejmattson.discordkt.api.dsl.listeners
+import me.jakejmattson.discordkt.api.extensions.toSnowflake
+import me.moeszyslak.logbot.dataclasses.Configuration
+import me.moeszyslak.logbot.embeds.createVoiceJoinEmbed
+import me.moeszyslak.logbot.embeds.createVoiceLeaveEmbed
+
+fun voiceListener(configuration: Configuration) = listeners {
+    on<VoiceStateUpdateEvent> {
+        val guildId = state.guildId ?: return@on
+        val guildConfig = configuration[guildId.longValue] ?: return@on
+
+        val channel = kord.getChannelOf<TextChannel>(guildConfig.historyChannel.toSnowflake())
```

Comment:
I think this would be better in the log channel rather than history

Reply:
👍 done